src/Corax/Indexing/IndexWriter.cs

```diff
@@ -612,41 +612,38 @@ private bool TryDeleteEntry(Slice termSlice, out long entryId)
             if (_indexedEntries.Contains(termSlice) == false)
             {
                 _compactKeyScope.Key.Set(termSlice);
-                //
                 var exists = _fieldsTree.CompactTreeFor(_fieldsMapping.GetByFieldId(0).FieldName).TryGetValue(_compactKeyScope.Key, out var containerId);
                 if (exists)
                 {
                     // note that the containerId may be a single value or many(!), if it is many items
                     // we'll delete them, but treat this as a _new_ entry, not an update to an existing
                     // one
-                    return RecordDeletion(containerId, out entryId);
+                    var result = RecordDeletion(containerId, out entryId, out var setsAreDisjoint);
+                    Debug.Assert(setsAreDisjoint);
```

Comment:
Is it valid assertion?

Reply:
Please add a message as the second parameter

Reply:
Yes, we do not expect to have common elements between the scheduled removal  list and posting list.